tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -398,6 +398,48 @@ static struct device_info boards[] = {
 		.first_sysupgrade_partition = "os-image",
 		.last_sysupgrade_partition = "file-system",
 	},
+	
+	/** Firmware layout for the C2v3 */
+	{
+		.id = "ARCHER-C2-V3",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:ArcherC2,product_ver:3.0.0,special_id:00000000}\n"
+			"{product_name:ArcherC2,product_ver:3.0.0,special_id:55530000}\n"
+			"{product_name:ArcherC2,product_ver:3.0.0,special_id:45550000}\n",
+		.support_trail = '\x00',
+		.soft_ver = "soft_ver:3.0.1\n",
+
+		/**
+		    We use a bigger os-image partition than the stock images (and thus
+		    smaller file-system), as our kernel doesn't fit in the stock firmware's
+		    1MB os-image.
+		*/
+		.partitions = {
+			{"factory-boot", 0x00000, 0x20000},
+			{"fs-uboot", 0x20000, 0x10000},
+			{"os-image", 0x30000, 0x200000},	/* Stock: base 0x30000 size 0x100000 */
+			{"file-system", 0x230000, 0x5A0000},	/* Stock: base 0x130000 size 0x6a0000 */
```

Comment:
It would be great if this device is using dynamic partitioning, which tplink-safeloader supports since this commit: https://github.com/openwrt/openwrt/commit/408c54b02f6a6dd2c0ae30a7885a37c9ac4c1b53